src/services/PeerDatabase.ts

```diff
@@ -48,22 +48,23 @@ export default class PeerDatabase {
       if (peer === this.identity.getUri()) {
         continue
       }
-      try {
-        // Check for invalid peer addresses. Validate peer only if not in set.
-        if (!this.peers.has(peer) && shallowValidatePeer(peer)) {
+      // Check for invalid peer addresses. Validate peer only if not in set.
+      if (!this.peers.has(peer) && shallowValidatePeer(peer)) {
+        try {
           const peerInfo = await axios.get(peer + '/info')
           if (peer === peerInfo.data.uri) {
             this.memoryMap.set(peer, peerInfo.data.fullMem)
             this.peers.add(peer)
           }
-        }
-
-      } catch (e) {
-        if (process.env.NODE_ENV !== 'test') {
-	        log.error('%s for %s', e, peer + '/info')
+        } catch (e) {
+          if (process.env.NODE_ENV !== 'test') {
+            log.debug('%s for %s', e, peer + '/info')
+          }
+          if (e.response && e.response.status === 404) {
```

Comment:
Is checking for a `404` not ambiguous? It could be possible that the server isn't running a Codius host anymore and that would still return a `404`. What about falling back on a previous endpoint like `/version`?

Reply:
I think this is thinking about this too much.
We will find out whether they are actually running a codius node when we actually try to talk to them. This is just collecting a peer list from the gossip network and we shouldn't be actively validating every single peer we hear about or we will destroy the network. We don't do any validation currently and I don't think we should beyond sanity checking for totally invalid addresses. There is no reason for a codiusd node to need to know another node's information beyond it's address. This all needs to be re-written anyway when adding in the peer-finder lib so this is just a stop-gap fix to allow us to release a new version and not immediately break the gossip network.